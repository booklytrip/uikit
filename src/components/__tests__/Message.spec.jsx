import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Message, MessageHeader } from '../Message';

describe('Message', () => {
    it('renders without crashing', () => {
        mount(<Message />)
    });

    describe('Snapshots', () => {
        it('renders with custom size', () => {
            const rendered = renderer.create(<Message size="small">Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with custom color', () => {
            const rendered = renderer.create(<Message color="orange">Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders of basic type', () => {
            const rendered = renderer.create(<Message basic>Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders of warning type', () => {
            const rendered = renderer.create(<Message warning>Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders of info type', () => {
            const rendered = renderer.create(<Message info>Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders of error type', () => {
            const rendered = renderer.create(<Message error>Message</Message>);
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders with header', () => {
            const rendered = renderer.create(
                <Message>
                    <MessageHeader>Header</MessageHeader>
                    Message
                </Message>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
