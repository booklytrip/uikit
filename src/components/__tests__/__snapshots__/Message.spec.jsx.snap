exports[`Message Snapshots renders of basic type 1`] = `
<div
  className="ui message"
  style={
    Object {
      "backgroundColor": "transparent",
      "boxShadow": "none",
    }
  }>
  Message
</div>
`;

exports[`Message Snapshots renders of error type 1`] = `
<div
  className="ui message error"
  style={Object {}}>
  Message
</div>
`;

exports[`Message Snapshots renders of info type 1`] = `
<div
  className="ui message info"
  style={Object {}}>
  Message
</div>
`;

exports[`Message Snapshots renders of warning type 1`] = `
<div
  className="ui message warning"
  style={Object {}}>
  Message
</div>
`;

exports[`Message Snapshots renders with custom color 1`] = `
<div
  className="ui message orange"
  style={Object {}}>
  Message
</div>
`;

exports[`Message Snapshots renders with custom size 1`] = `
<div
  className="ui message small"
  style={Object {}}>
  Message
</div>
`;

exports[`Message Snapshots renders with header 1`] = `
<div
  className="ui message"
  style={Object {}}>
  <div
    className="header">
    Header
  </div>
  Message
</div>
`;
